exercises/practice/protein-translation/.meta/example.rkt

```diff
@@ -0,0 +1,24 @@
+#lang racket
+
+(provide proteins)
+
+(define (translate protein)
+  (case protein
+    [("AUG") "Methionine"]
+    [("UUU" "UUC") "Phenylalanine"]
+    [("UUA" "UUG") "Leucine"]
+    [("UCU" "UCC" "UCA" "UCG") "Serine"]
+    [("UAU" "UAC") "Tyrosine"]
+    [("UGU" "UGC") "Cysteine"]
+    [("UGG") "Tryptophan"]
+    [("UAA" "UAG" "UGA") '()]
+    [else (error "Invalid codon")]))
+
+(define (proteins strand)
+  (cond
+    [(zero? (string-length strand)) '()]
+    [(< (string-length strand) 3) (error "Incomplete RNA sequence can't translate")]
```

Comment:
This works perfectly as-is for passing the CI, but if I were solving this, I'd define a contract on translate so it can only take codons that are exactly three letters long. This is a totally optional tweak, but if you do go that route, it'll be a bit cleaner if you return an empty string and not an empty list on line 14.

Reply:
I guess I could return "STOP" instead of the empty list.